src/thunderbots/software/geom/triangle.cpp

```diff
@@ -0,0 +1,35 @@
+#include "geom/triangle.h"
+
+#include <algorithm>
+#include <unordered_set>
+
+#include "geom/util.h"
+
+Triangle::Triangle(Point p0, Point p1, Point p2) : Polygon(std::vector<Point>{p0, p1, p2})
+{
+    segments[0] = Segment{p0, p1};
```

Comment:
You can't assign elements of the vector without allocating / initializing them first. This is why the test are failing, because this causes a segfault. The memory this segment is being assigned to hasn't been allocated yet.

This should probably use `emplace_back` instead to append the value to the vector. This goes for the other functions in this file too that are setting up `segments`